internal/nodes/avi_model_l4_translator.go

```diff
@@ -385,3 +389,43 @@ func GetDefaultSubDomain() []string {
 	}
 	return cloudProperty.NSIpamDNS
 }
+
+func buildL4InfraSetting(vs *AviVsNode, vsvip *AviVSVIPNode, svc *corev1.Service, advl4GWClassName *string) {
+	var err error
+	var infraSetting *akov1alpha1.AviInfraSetting
+
+	if lib.UseServicesAPI() && advl4GWClassName != nil {
+		gwClass, err := lib.GetSvcAPIInformers().GatewayClassInformer.Lister().Get(*advl4GWClassName)
+		if err != nil {
+			utils.AviLog.Warnf("Unable to get corresponding GatewayClass %s", err.Error())
+			return
+		} else {
+			if gwClass.Spec.ParametersRef != nil && gwClass.Spec.ParametersRef.Group == lib.AkoGroup && gwClass.Spec.ParametersRef.Kind == lib.AviInfraSetting {
+				infraSetting, err = lib.GetCRDInformers().AviInfraSettingInformer.Lister().Get(gwClass.Spec.ParametersRef.Name)
+				if err != nil {
+					utils.AviLog.Warnf("Unable to get corresponding AviInfraSetting: %s", err.Error())
+					return
+				}
+			}
+		}
+	} else if infraSettingAnnotation, ok := svc.GetAnnotations()[lib.InfraSettingNameAnnotation]; ok && infraSettingAnnotation != "" {
+		infraSetting, err = lib.GetCRDInformers().AviInfraSettingInformer.Lister().Get(infraSettingAnnotation)
+		if err != nil {
+			utils.AviLog.Warnf("Unable to get corresponding AviInfraSetting: %s", err.Error())
```

Comment:
both the warns on line 406 and 414 are identical. I guess you will have to rely on line numbers to figure out if the annotations failed or the gateway class didn't have the CRD reference. Maybe it's a good idea to distinguish the two?

Reply:
yes, done. passed on the key as well